### DISCUSSION TITLE
How to obtain DataContext in OS browser

### DISCUSSION THREAD

**Discussion by fanaone** [AvaloniaUI/Avalonia#13586] [Q&A]
Click on the Grid control in the DataTemplate within TabControl, and respond to the following event after clicking Release
点击TabControl内DataTemplate中的Grid控件，点击释放后响应以下事件

The DataContext of the grid control can be obtained on the OS Windows platform; But getting null in OS browser
在os windows平台能获取到 grid 控件的 DataContext；但是在 os browser 中获取为null

How to obtain DataContext in OS browser
如何在浏览器项目中获取DataContext
```C#
 
    private async void InputElement_OnPointerReleased(object? sender, PointerReleasedEventArgs e)
    {      
        var obj = sender as Grid;
        var dc = obj.DataContext as TabItemModel;
        
        Console.WriteLine(dc?.ToString() ?? "null");
        baseDc.TabItems.Remove(dc);
    }
```

**Comment by Coloryr** [comment]
浏览器还是用其他框架写吧，这玩意多少有点问题的

**Comment by maxkatz6** [comment]
There shouldn't be any difference in data context between browser and desktop. 
But you need to keep in mind, that MainWindow is not used on Browser. In your `App.axaml.cs` you can see that you have MainView or MainWindow used depending on the supported lifetime (browser doesn't support windows).

**Comment by fanaone** [comment]
I can obtain the `SelectedIndex `of `TabControl`; Then locate the bound collection through `SelectedIndex`
我可以获得TabControl的SelectedIndex；然后通过SelectedIndex定位绑定集合
